### ISSUE TITLE
Creating a Github organization?

### ISSUE THREAD

**Issue by cpitclaudel** [realgud/realgud#112]
In the spirit of rewriting the README and adding a screenshot to improve our visibility and ease-of-use, what would you think of creating a Github organization for this project? RealGUD and the related emacs packages could be moved there, and we'd keep the existing stars, issues, and everything (GitHub just transparently sets up redirects). This would make it easier to have external contributors, too (for example, I'd get access to editing labels)

Plus, we could get a logo! Something like this maybe? (did I get the lower/uppercase right?)

![realgud](https://cloud.githubusercontent.com/assets/8181630/15658463/c29204fe-268a-11e6-8c42-a2459ee3d2b0.png)

Which we could strip down to the following to use as a square icon:

![realgud-square](https://cloud.githubusercontent.com/assets/8181630/15658477/efa53aba-268a-11e6-9351-3c8ad4e60395.png)

(it's supposed to evoke a breakpoint).

What do you think? And @amelio-vazquez-reina, do you have an opinion on this logo?

**Comment by amelio-vazquez-reina** [comment]
I love it! Nicely done @cpitclaudel I personally prefer the first one with the package name spelled out; It will help people remember the name of the package and spread the word about it.

Frankly, I think this effort has huge potential and that not enough people have heard of it yet, so the new `README.md` and logo will make a big difference. I'm all for supporting this effort in any way possible (GitHub organization, etc). Nice job again! 👍

**Comment by cpitclaudel** [comment]
Thanks for the feedback @amelio-vazquez-reina :) I think we'd use the fully spelled-out one in most places, but as an avatar for a github organization the square one would work nicely, I think :)

**Comment by amelio-vazquez-reina** [comment]
Oh perfect. Yes, that makes sense 👍

**Comment by rocky** [comment]
I'm okay with having this move to an organization. BTW we haven't heard back from @purcell ll yet regarding the autoloads. A repository move is something that will need coordination with Melpa. 

As for  logos it's just not something I feel like I have any expertise at so I leave it to others' judgement. If you want to use that and it helps or you think it helps, great!

I gather that when the project is transferred the bug tracker moves as well? The code and wikis are easily moved. Also, I guess I'd move those other external debuggers? 

And how about dependent programs like test-simple and loc-changes?

**Comment by cpitclaudel** [comment]
I think you're overestimating the amount of effort involved here :) Everything is moved automatically (including the issues, the wikis, the code), and redirects are put in place so checkouts and links using the old URLs remain valid indefinitely in the future.

We'd want to update the MELPA recipe for good measure, but there'd be no particular coordination issues: it would be no problem for MELPA to keep referring to the old URL until we update the recipe.

https://help.github.com/articles/transferring-a-repository/#transferring-from-a-user-to-an-organization has more details.

As for what to move, I don't think the external debuggers should be moved. They are independent projects, with no direct connection to Emacs, and no dependency on RealGUD (except for the fact that RealGUD offers a convenient way to use them). Other Elisp packages that add support for other debuggers in RealGUD are probably good candidates for a move as well, but not so much loc-changes and test-simple: they are also independent programs that have applicability beyond RealGUD, I think.

Thus concretely my proposal is to

* Create `realgud` organization on GitHub
* Migrate  realgud, realgud-byebug, realgud-old-debuggers, realgud-pry, and realgud-rdb2 to it.

The model that I'm following here is that of Flycheck, (https://github.com/flycheck/). I can take care of creating the organization, if you want :)

Re logos: neat :)

**Comment by rocky** [comment]
Ok. I'll look at setting up the organization a bit later.

**Comment by cpitclaudel** [comment]
Neat :)

**Comment by cpitclaudel** [comment]
Just accepted the invitation, thanks! Exciting stuff happening. (Just pushed a small fix to the square logo, too, but I can't update it myself unless you make me a joint 'owner' of the organization)

**Comment by rocky** [comment]
Done. I want to wait on the main repo for a day to see what happens on melpa on these. And I may rename the repo to just realgud down the line  too.

**Comment by cpitclaudel** [comment]
Awesome :) Yes, I think waiting to see if MELPA catches up properly sounds wise. Renaming the repo does too; I was expecting something like http://github.com/realgud/realgud; that would be very good.

**Comment by purcell** [comment]
I was going to comment that checking in an autoloads file isn't a great solution IMO, but it looked like the issue where you'd previously tagged me had been closed, and the associated change later reverted, so I figured you'd all come to a decision without my input.

Re. transferring the repo, github redirects everything nicely, so there'd be no hurry to update the MELPA recipe.

**Comment by rocky** [comment]
@purcell we hadn't come to any conclusion. We were instead waiting to get your view. 

If checking in an autoload file isn't a great solution, what are the other possibilities? 

Allow MELPA to run a hook script in the repo like @cpitclaudel created to generate the autoloads?

**Comment by rocky** [comment]
BTW If there's not a hitch in transferring and renaming the repo, I'll do that soon.

**Comment by cpitclaudel** [comment]
@purcell: Indeed, we hadn't come to a conclusion :) We were waiting for your advice.

Checking in an autoloads file still seems like the best solution to me. Asking on emacs-dev didn't produce better options, and the alternative that consists in dynamically generating those by running custom autoloads generation code at compile time seems error prone to me. 

What do you think?

**Comment by purcell** [comment]
Having trouble finding time to read around this and then have an opinion, sorry. :-(

**Comment by cpitclaudel** [comment]
@purcell No worries :) Thanks for all your hard work on MELPA!

**Comment by cpitclaudel** [comment]
@rocky I'd be of the opinion of moving forward with this. The lack of autoloads is a pain for our users (they have to remember to `load-library`), and the solution is fairly unintrusive and emacs-devel backed. I'll prepare a new PR, and we can make the final call there :)

**Comment by rocky** [comment]
Ok. Merged back in. Thanks @cpitclaudel for doing this and @purcell for giving guidance here.
